code/game/objects/machinery/minelayer.dm

```diff
@@ -0,0 +1,87 @@
+/obj/item/minelayer
+	name = "\improper M21 APRDS \"Minelayer\""
+	desc = "Anti-Personnel Rapid Deploy System, APRDS for short, is a device designed to quickly deploy M20 mines in large quantities. WARNING: Operating in tight places or existing mine fields will result in reduced efficiency."
+	icon = 'icons/Marine/marine-items.dmi'
+	icon_state = "minelayer"
+
+	max_integrity = 200
+	flags_item = IS_DEPLOYABLE|DEPLOYED_NO_ROTATE
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/minelayer/Initialize()
+	. = ..()
+	AddElement(/datum/element/deployable_item, /obj/machinery/deployable/minelayer, 1 SECONDS)
+
+/obj/machinery/deployable/minelayer
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	use_power = NO_POWER_USE
+	///how many mines are currently stored
+	var/stored_amount = 0
+	///how many mines can be stored
+	var/max_amount = 10
+	///radius on mine placement
+	var/range = 4
+
+/obj/machinery/deployable/minelayer/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	playsound(loc, 'sound/machines/click.ogg', 25, 1)
+
+	sleep(2 SECONDS)
+
+	var/list/turf_list = list()
+
+	for(var/turf/T AS in orange(range, loc))
+		turf_list += T
+
+	while(stored_amount > 0 && turf_list.len > 0)
+		var/turf/target_turf = pick_n_take(turf_list)
+		if(!target_turf.density && !turf_block_check(src, target_turf) && !(locate(/obj/item/explosive/mine) in range(1, target_turf)))
```

Comment:
i dislike the amount of range checks this creates but I dont see a better solution easily right now